src/test/java/spoon/test/filters/FilterTest.java

```diff
@@ -115,9 +115,17 @@ public void testNameFilter() throws Exception {
 		CtClass<?> foo = factory.Package().get("spoon.test.filters.testclasses").getType("Foo");
 		assertEquals("Foo", foo.getSimpleName());
 		List<CtNamedElement> elements = foo.getElements(new NameFilter<>("i"));
+		// contract: tests NameFilter constructor for cases other than null
 		assertEquals(1, elements.size());
+		// contract: tests NameFilter constructor for null value case
+		assertThrows(IllegalArgumentException.class, () -> foo.getElements(new NameFilter<>(null)));
```

Comment:
Calling `foo.getElements` is redundant as `new NameFilter<>(null)` already throws.

```suggestion
		assertThrows(IllegalArgumentException.class, () -> new NameFilter<>(null));
```

Reply:
Yes, it is actually redundant, thanks for pointing it out I welcome it.